File: Tutoria1/Tutoria1/ViewController.m
//
//  ViewController.m
//  Tutoria1
//
//  Created by Nicolás Gebauer on 20-03-15.
//  Copyright (c) 2015 Nicolás Gebauer. All rights reserved.
//

#import "ViewController.h"
#import "DelegateTableView.h"
#import "DetailsViewController.h"

@interface ViewController ()

@property (nonatomic, strong) DelegateTableView* tabla;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    _tabla = [[DelegateTableView alloc]init];
    _tabla.referenciaViewController = self;
    _TablaTutoria.delegate = _tabla;
    _TablaTutoria.dataSource = _tabla;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender{
    DetailsViewController *controller = segue.destinationViewController;
    controller.stringPersonalizado = [NSString stringWithFormat:@"La celda apretada es la: %ld", _tabla.numeroCelda ];
}

@end

File: IoPTarea2/IoPTarea2/ViewController.swift
//
//  ViewController.swift
//  IoPTarea2
//
//  Created by Nicolás Gebauer on 25-03-15.
//  Copyright (c) 2015 Nicolás Gebauer. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    @IBOutlet weak var TablaContactos: UITableView!
    var delegateTableView : DelegateTableView?

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        delegateTableView = DelegateTableView()
        delegateTableView!.armarListaBonita()
        delegateTableView!.referenciaAlControlador = self
        TablaContactos.dataSource = delegateTableView
        TablaContactos.delegate = delegateTableView
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        let controllerDetalles = segue.destinationViewController as DetailsLista
        controllerDetalles.stringDetalles = delegateTableView!.listaAlumnos[delegateTableView!.numeroCelda]
    }
}
